### PR TITLE
FE-1128 - Batch #1 of UX and Visual Enhancements Across Features 

### PR DESCRIPTION
**Pull request by nicklemmon**
[FE-1128](https://jira.int.messagesystems.com/browse/FE-1128)

### What Changed

- Removed the copy icon from the `CopyField` component
- Incorporate `<PowerSettingsNew/>` icon in to the enable/disable SSO button
- Fixes positioning problems on the alert details pages by removing the use of `<Box/>` and adjusting custom styles
- Fixed some button positioning problems on the alert details page
- Within the RV tags on `/recipient-validation/list`, fixed a lot of inconsistent spacing issues caused by one-off/custom CSS
- Removed some unhelpful snapshot tests
- Hibana Cypress tests now run in parallel with OG theme integration tests in CircleCI

### How To Test
- [x] Go to `/reports/message-events` and click on the 'Share' button - verify no icon renders within the copy button inside the modal
- [x] Go to `/account/settings` and verify that the `<PowerSettingsNew/>` icon renders within the Enable and Disable SSO buttons
- [x] Go to `/alerts/details/<id>` and verify button positioning is correct, and not flush right with the containing `Panel`
- [x] Go to `/recipient-validation/list` and verify the tabs work as expected (i.e., uploading a file, doing a single validation, rendering API details). Also verify that spacing/component use is consistent visually
- [x] Verify that Hibana Cypress tests run simultaneously with OG tests

### To Do
- [ ] Incorporate feedback

### CODE REVIEW

File: src/pages/alerts/components/AlertDetails.js [hunk 1]
```diff
@@ -209,7 +210,10 @@ export const AlertDetails = ({
   return (
     <Panel data-id="alert-details-panel">
       <SectionComponent paddingLeft="500" paddingTop="400" width="100%">
-        <h3 className={styles.Title}>Alert Details</h3>
+        <Heading as="h3" className={styles.Title}>
+          Alert Details
+        </Heading>
```
Comment (nicklemmon) [inline src/pages/alerts/components/AlertDetails.js:215]: Just updated this while I was here.

File: src/pages/alerts/components/AlertDetailsHibana.module.scss [hunk 2]
```diff
@@ -1,7 +1,7 @@
 @import '~@sparkpost/design-tokens-hibana/tokens';
 
 .Title {
-  top: spacing(200);
+  margin-bottom: spacing(500);
```
Comment (nicklemmon) [inline src/pages/alerts/components/AlertDetailsHibana.module.scss:4]: The old style didn't seem to be doing anything

File: src/pages/recipientValidation/components/SingleAddressForm.js [hunk 3]
```diff
@@ -2,30 +2,34 @@ import React, { Component } from 'react';
...
-  singleAddressBody = () => {
+  render() {
     const { styles } = this.props;
 
     return (
       <>
-        <div className={classNames(styles.Header)}>Validate a Single Address</div>
+        <Heading as="h3" looksLike="h5" className={styles.Header}>
+          Validate a Single Address
+        </Heading>
```
Comment (nicklemmon) [inline src/pages/recipientValidation/components/SingleAddressForm.js:27]: Improved HTML semantics while I was here

File: src/pages/alerts/components/AlertIncidents.js [hunk 4]
```diff
@@ -62,7 +63,7 @@ const AlertIncidents = ({ incidents = [], alert, subaccountIdToString }) => {
         if (!tag) {
           return acc;
         }
-        return [...acc, <span key={'join_and'}> and </span>, tag];
+        return [...acc, <span key="join_and"> and </span>, tag];
```
Comment (nicklemmon) [inline src/pages/alerts/components/AlertIncidents.js:66]: Just a Prettier change here

File: cypress/tests/integration/signals-analytics/analytics-report/constants/index.js [hunk 5]
```diff
@@ -107,11 +107,11 @@ export const METRICS = [
     queryParam: 'count_hard_bounce',
   },
   {
-    name: 'In-band Bounce',
+    name: 'In-band Bounces',
     queryParam: 'count_inband_bounce',
   },
   {
-    name: 'Out-of-band Bounce',
+    name: 'Out-of-band Bounces',
```
Comment (nicklemmon) [inline cypress/tests/integration/signals-analytics/analytics-report/constants/index.js:114]: Typos were causing test failures

File: cypress/tests/integration/signals-analytics/engagement/engagementReportPage.spec.js [hunk 6]
```diff
@@ -12,288 +12,290 @@ const STABLE_UNIX_DATE = 1581087062000; // Stable unix timestamp (2/6/2020)
...
-describe('The engagement report page', () => {
-  beforeEach(() => {
-    cy.stubAuth();
-    cy.login({ isStubbed: true });
-
-    cy.stubRequest({
-      url: '/api/v1/subaccounts',
-      fixture: 'subaccounts/200.get.json',
-      requestAlias: 'subaccountsRequest',
+if (Cypress.env('DEFAULT_TO_HIBANA') !== true) {
```
Comment (nicklemmon) [inline cypress/tests/integration/signals-analytics/engagement/engagementReportPage.spec.js:15]: All I did was add this conditional check

Comment (nicklemmon) [reply cypress/tests/integration/signals-analytics/engagement/engagementReportPage.spec.js:15]: Many of the tests in this file remain commented out, sadly, due to a problem with how Cypress handles requests from similarly-structured URLs

File: src/pages/recipientValidation/tests/RecipientValidationPage.test.js [hunk 7]
```diff
@@ -57,7 +57,7 @@ describe('Page: Recipient Email Verification (shallow)', () => {
 
   it('renders Api tab correctly when selected', () => {
     wrapper = subject({ tab: 2 });
-    expect(wrapper.find('ApiIntegrationDocs')).toExist();
+    expect(wrapper.find('ApiDetails')).toExist();
```
Comment (nicklemmon) [inline src/pages/recipientValidation/tests/RecipientValidationPage.test.js:60]: This component was just renamed

File: src/components/copyField/CopyField.js [hunk 8]
```diff
@@ -50,10 +49,7 @@ class CopyField extends Component {
           disabled={!copied}
         >
           <Button variant="connected" name="copy-field-button" onClick={this.handleCopy}>
-            <Box as="span" mr="100">
-              <ContentCopy size={14} />
```
Comment (bkemper) [inline src/components/copyField/CopyField.js:54]: Should we use this component in CopyField?

File: src/pages/account/components/SingleSignOnPanel/StatusSection.js [hunk 9]
```diff
@@ -80,7 +81,19 @@ export class StatusSection extends React.Component {
         actions={[
           {
             color: 'orange',
-            content: enabled ? 'Disable SSO' : 'Enable SSO',
+            content: enabled ? (
+              <div className={styles.SSOButton}>
```
Comment (bkemper) [inline src/pages/account/components/SingleSignOnPanel/StatusSection.js:85]: This puts a `<div>` inside a `<button>` and `<span>`.  Is there a better way around this?  Can we make this a `<span style="display: inline-flex">`?

Screen Shot 2020-07-16 at 2 07 58 PM

File: src/pages/account/components/SingleSignOnPanel/tests/__snapshots__/StatusSection.test.js.snap [hunk 10]
```diff
@@ -6,7 +6,16 @@ exports[`StatusSection disables buttons when updating 1`] = `
     Array [
       Object {
         "color": "orange",
-        "content": "Enable SSO",
+        "content": <div
```
Comment (bkemper) [inline src/pages/account/components/SingleSignOnPanel/tests/__snapshots__/StatusSection.test.js.snap:9]: 😄

File: src/pages/alerts/components/tests/AlertDetails.test.js [hunk 11]
```diff
@@ -28,11 +28,6 @@ describe('Alert Details Component', () => {
     renderPrimaryAreaComponent: () => <div>My Primary Area</div>,
   };
 
-  it('should render the alert details component correctly', () => {
-    const wrapper = shallow(<AlertDetails {...props} />);
-    expect(wrapper).toMatchSnapshot();
```
Comment (bkemper) [inline src/pages/alerts/components/tests/AlertDetails.test.js:33]: 👏

File: src/pages/account/components/SingleSignOnPanel/StatusSection.js (hunk 9 above)
Comment (nicklemmon) [reply src/pages/account/components/SingleSignOnPanel/StatusSection.js:85]: oooh that is weird - yeah, let me re-adjust that.

File: src/components/copyField/CopyField.js (hunk 8 above)
Comment (nicklemmon) [reply src/components/copyField/CopyField.js:54]: Not sure what you mean? Which component?

Comment (bkemper) [reply src/components/copyField/CopyField.js:54]: 😊 I was looking at the source when typing this comment.  Let me try again...

CopyField has almost the identical code as CopyToClipboard.  Should we just use CopyToClipboard in CopyField?
